CHANGELOG.md

```diff
@@ -10,6 +10,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Allow mmap to use new JDK-19 preview APIs in Apache Lucene 9.4+ ([#5151](https://github.com/opensearch-project/OpenSearch/pull/5151))
 - Add events correlation engine plugin ([#6854](https://github.com/opensearch-project/OpenSearch/issues/6854))
 - Add support for ignoring missing Javadoc on generated code using annotation ([#7604](https://github.com/opensearch-project/OpenSearch/pull/7604))
+- Added server version as response header [#6583](https://github.com/opensearch-project/OpenSearch/issues/6583)
```

Comment:
This should go in the `Unreleased 2.x` section as this should be backported to the 2.x line. @reta correct me if I'm wrong please.

Also, I'm a pedantic nitpick :) so please write this in the imperative like the others...e.g. "Add server version as REST response header"

Reply:
Moved and written in imperative